pkg/didcomm/protocol/issuecredential/states_test.go

```diff
@@ -0,0 +1,287 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package issuecredential
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func notTransition(t *testing.T, st state) {
+	t.Helper()
+
+	var allState = [...]state{
+		// common states
+		&start{}, &abandoning{}, &done{}, &noOp{},
+		// states for Issuer
+		&proposalReceived{}, &offerSent{}, &requestReceived{}, &credentialIssued{},
+		// states for Holder
+		&proposalSent{}, &offerReceived{}, &requestSent{}, &credentialReceived{},
+	}
+
+	for _, s := range allState {
+		require.False(t, st.CanTransitionTo(s))
+	}
+}
+
+func TestStart_CanTransitionTo(t *testing.T) {
+	st := &start{}
+	require.Equal(t, stateNameStart, st.Name())
+
+	// common states
+	require.False(t, st.CanTransitionTo(&start{}))
+	require.False(t, st.CanTransitionTo(&abandoning{}))
+	require.False(t, st.CanTransitionTo(&done{}))
+	require.False(t, st.CanTransitionTo(&noOp{}))
+
+	// states for Issuer
+	require.True(t, st.CanTransitionTo(&proposalReceived{}))
+	require.False(t, st.CanTransitionTo(&offerSent{}))
+	require.False(t, st.CanTransitionTo(&requestReceived{}))
+	require.False(t, st.CanTransitionTo(&credentialIssued{}))
+
+	// states for Holder
+	require.True(t, st.CanTransitionTo(&proposalSent{}))
+	require.False(t, st.CanTransitionTo(&offerReceived{}))
+	require.False(t, st.CanTransitionTo(&requestSent{}))
+	require.False(t, st.CanTransitionTo(&credentialReceived{}))
+}
+
+func TestStart_ExecuteInbound(t *testing.T) {
+	followup, action, err := (&start{}).ExecuteInbound(&metaData{})
+	require.Error(t, err)
```

Comment:
Normally you should check for specific error message but since this is 'not implemented' it is ok to leave it as is.

Reply:
Done!